patroni/zookeeper.py

```diff
@@ -220,6 +220,11 @@ def delete_leader(self):
         if isinstance(self.leader, Leader) and self.leader.name == self._name:
             self.client.delete(self.client_path('/leader'))
 
+    def cancel_initialization(self):
+        node = self.get_node(self.initialize_key)
+        if node and node == self._name:
+            self.client.delete(self.client_path(self.initialize_key))
```

Comment:
If you want to really be sure that you are removing the right node, you should also pass version=znode.mzxid as an argument